uniffi_core/src/ffi/rustcalls.rs

```diff
@@ -42,52 +42,42 @@ use std::panic;
 #[repr(C)]
 pub struct RustCallStatus {
     pub code: RustCallStatusCode,
-    // code is signed because unsigned types are experimental in Kotlin
-    pub error_buf: MaybeUninit<RustBuffer>,
-    // error_buf is MaybeUninit to avoid dropping the value that the consumer code sends in:
-    //   - Consumers should send in a zeroed out RustBuffer.  In this case dropping is a no-op and
-    //     avoiding the drop is a small optimization.
-    //   - If consumers pass in invalid data, then we should avoid trying to drop it.  In
-    //     particular, we don't want to try to free any data the consumer has allocated.
-    //
-    // `MaybeUninit` requires unsafe code, since we are preventing rust from dropping the value.
-    // To use this safely we need to make sure that no code paths set this twice, since that will
-    // leak the first `RustBuffer`.
+    pub error_buf: ManuallyDrop<RustBuffer>,
+    // error_buf is owned by the foreign side.
+    // - Whatever we are passed, we must never free.
```

Comment:
I think the foreign side always passes in an empty buffer, this is important since otherwise we would leak it whenever we write to the field.